client/src/containers/HomePage/HomePage.jsx

```diff
@@ -148,16 +150,40 @@ class HomePage extends Component {
   }
 
   handleJoyrideCallback = data => {
-    const { action, index, status, type  } = data;
+    const { action, index, status, type } = data;
+   /*
+   The issue -> how to manage it so that at the beginning of step 5, it will send you back to the pie charts when you click "next" without the table being open
+   -> but, at the end, it is able to mount and skip to the next step when the table is closed 
+   -> I could make it so that clicking next automatically closes the table and skips to the next step
+   
+   */
 
-    if ([EVENTS.TARGET_NOT_FOUND].includes(type)) {
+    console.log(this.state.currentDisplay);
+
+    // if the index == 4 and you're at step after and 
+    // special cases -> if ladder?
+
+    if (this.state.skipStep && type == EVENTS.STEP_BEFORE) {
+      this.setState({stepIndex: 4, skipStep: false});
+    }
+    else if (index == 4 && [EVENTS.STEP_AFTER].includes(type) && !this.chartsRef.current.state.dialogOpen) {
+        this.setState({stepIndex: index - 1, skipStep: true});
+    } else if ((index == 3 &&  this.state.currentDisplay !== 'California') || 
```

Comment:
Let's make it in chronological order of index for readability. You could also separate it into another isReadyToStep() function (inverted) if you don't like the big OR chain. The only thing you'd need to pass in is type.

Could also define constants for readability. JOYRIDE_INDEX_CLICK_STATE = 3, JOYRIDE_LOCK_STATE = 'California'